src/components/Button/Button.tsx

```diff
@@ -104,21 +114,27 @@ class Button extends UIComponent<any, any> {
         {...accessibility.attributes.root}
         {...rest}
       >
-        {hasChildren && children}
-        {!hasChildren && iconPosition !== 'after' && this.renderIcon()}
-        {!hasChildren && content}
-        {!hasChildren && iconPosition === 'after' && this.renderIcon()}
+        {iconPosition !== 'after' && this.renderIcon(variables)}
+        {content && <span className={classes.content}>{content}</span>}
+        {iconPosition === 'after' && this.renderIcon(variables)}
       </ElementType>
     )
   }
 
-  public renderIcon = () => {
+  public renderIcon = variables => {
     const { disabled, icon, iconPosition, content, type } = this.props
 
     return Icon.create(icon, {
       defaultProps: {
         xSpacing: !content ? 'none' : iconPosition === 'after' ? 'before' : 'after',
-        variables: { color: type === 'primary' && !disabled ? 'white' : undefined },
+        variables: {
+          color:
+            type === 'primary'
+              ? variables.typePrimaryColor
+              : type === 'secondary'
+                ? variables.typeSecondaryColor
+                : variables.color,
+        },
```

Comment:
yes, agree with the reasons provided above, thanks - this goes in line with the exactly the same discussion we've had with Miro. For the sake of avoiding to introduce additional complexity at that moment we've decided to move forward with the simplest approach that would **indicate our intent**, and after that change it in a way that theming aspects will be completely separated from logic.

Speaking of the options suggested - I would definitely support the first one, as it suggest clear path that not changes but extends approach we are using now - and, thus, it is not associated with significant complexity increase. Also, with this one we will be able to address quite broad set of scenarios we are currently struggling with (or using the approach where variables are coupled with implementation)

Not saying 'no' to the second option, though - but would rather wait once this concept will solidify in our minds, for the sake of us being absolutely sure about what we are buying for additional efforts laid in this direction, as well as whether this concept is able to cover the cases we are interested in (and how flexible it would be to cover potentially emerged edge-cases)